drivers/couchbase/couchbase-driver/src/main/java/io/mongock/driver/couchbase/repository/CouchbaseRepositoryBase.java

```diff
@@ -0,0 +1,195 @@
+package io.mongock.driver.couchbase.repository;
+
+import com.couchbase.client.java.Cluster;
+import com.couchbase.client.java.Collection;
+import com.couchbase.client.java.json.JsonArray;
+import com.couchbase.client.java.json.JsonObject;
+import com.couchbase.client.java.manager.query.CreateQueryIndexOptions;
+import com.couchbase.client.java.manager.query.DropQueryIndexOptions;
+import com.couchbase.client.java.manager.query.QueryIndex;
+import com.couchbase.client.java.query.QueryOptions;
+import com.couchbase.client.java.query.QueryScanConsistency;
+import io.mongock.api.exception.MongockException;
+import io.mongock.driver.api.common.EntityRepository;
+import io.mongock.driver.api.common.RepositoryIndexable;
+import io.mongock.driver.couchbase.util.CollectionIdentifierUtil;
+import io.mongock.driver.couchbase.util.N1QLQueryProvider;
+import io.mongock.utils.field.FieldInstance;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Base Couchbase based entity repository. 
+ * To support both v6 and v7+ couchbase servers index actions are delegated to the index manager from cluster and collection respectively.
+ * 
+ * For more info on Scope-Level SQL++ Queries & all Collections features compatibility see https://docs.couchbase.com/java-sdk/current/project-docs/compatibility.html.
+ * 
+ * @author Tigran Babloyan
+ */
+public class CouchbaseRepositoryBase <DOMAIN_CLASS> implements EntityRepository<DOMAIN_CLASS, JsonObject>, RepositoryIndexable {
+  private final static Logger logger = LoggerFactory.getLogger(CouchbaseRepositoryBase.class);
+  private final static String INDEX_NAME = "idx_mongock_keys";
+  protected final static String DOCUMENT_TYPE = "mongock";
```

Comment:
This document type is currently used only for ChangeEntry. We suggest to rename it to "mongockChangeEntry" and add another one named "mongockLockEntry" for LockEntry documents (currenty LockEntry documents don't include "_type" field).